src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -1082,13 +1059,16 @@ internal static bool IsZeroElementTupleType(TypeSymbol type)
                 ParameterSymbol? parameter = null;
                 if (!isError)
                 {
+                    int parameterIndex = i + skippedExtensionParameters;
+                    if (parameterIndex < deconstructMethod!.ParameterCount)
+                    {
+                        parameter = deconstructMethod.Parameters[parameterIndex];
```

Comment:
Is this strictly related to moving to a ref safety visitor? Maybe this parameter symbol was missing in some cases where it was needed for the ref safety pass?

Reply:
> Maybe this parameter symbol was missing in some cases where it was needed for the ref safety pass?

Yes, `RefSafetyAnalysis.VisitPositionalSubpattern()` requires the `ParameterSymbol`.